lib/goals.js

```diff
@@ -1,23 +1,36 @@
-// Goal base class
+/**
+ * Goal base class
+ * @abstract
+ */
 class Goal {
-  // Return the distance between node and the goal
+  /**
+   * Return the distance between node and the goal
+   * @abstract
+   */
```

Comment:
I prefer // over /**/ for comments as it makes it easier to comment codes when developing.
I'm also not a big fan of bulky comments.